src/public_parameters/mem_cache.rs

```diff
@@ -72,8 +72,8 @@ impl PublicParamMemCache {
                     assert!(rest.is_empty());
                     Ok(Arc::new(pp.clone())) // this clone is VERY expensive
                 }
-                Err(e) => {
-                    eprintln!("{e}");
+                Err(_) => {
```

Comment:
It seems that failure to deserialize the public parameters is going to result in an `public_parameters::error::Error::CacheError`, could we be specific about only reacting to this?

Reply:
What do you mean?

Reply:
What is the type of `e`? are there variants of `e` for which we should NOT swallow the error?

Reply:
I think all errors are worth logging.
Here's an idea:
* `warn!` the error, regardless of its variant
* `info!` the message that new public parameters are being generated

Reply:
Done